docs/source/plugins.rst

```diff
@@ -0,0 +1,92 @@
+
+Writting plugins for **mokapot**
+================================
+
+This is a tutorial on how to write plugins for mokapot. 
+
+The plugin architecture allows you to extend mokapot without
+requiring any changes to the core of mokapot.
+
+Right not it allows you to modify the data before
+it gets used by the models or implement a new model.
+
+Plugins are python packages that export a `BasePlugin` object
+to an entry point called `mokapot.plugins`.
+So at thr very least you need to set that entry point up in your package configuration.
```

Comment:
**nitpik:** Rephrase this sentence. When writing technical documentation we want to avoid phrase like "at the very least," because it can feel like it trivializes a task that might be hard for some folks. 😉